src/install/install.zig

```diff
@@ -9669,6 +9669,25 @@ pub const PackageManager = struct {
             peer: bool = false,
         };
 
+        // std.process.args allocates!
+        const ArgsIterator = struct {
+            buf: [][:0]const u8,
+            i: u32 = 0,
+
+            pub fn next(this: *ArgsIterator) ?[]const u8 {
+                if (this.buf.len <= this.i) {
+                    return null;
+                }
+                const i = this.i;
+                this.i += 1;
+                return this.buf[i];
+            }
+
+            pub fn skip(this: *ArgsIterator) bool {
+                return this.next() != null;
+            }
+        };
+
```

Comment:
This `ArgsIterator` struct comes from [cli file](https://github.com/oven-sh/bun/blob/bun-v1.1.42/src/cli.zig#L1554-L1571) and maybe it should be extracted to be reused instead of copied but it is the first copy and code is pretty simple so I don't know if it is worth yet.